python/nano/src/bigdl/nano/pytorch/strategies/ipex/ipex_strategy.py

```diff
@@ -63,3 +76,47 @@ def setup(self, trainer: pl.Trainer) -> None:
             ipex.optimize(self.model, optimizer=self.optimizers[0], inplace=True, dtype=dtype)
         else:
             invalidInputError(False, "Ipex does not support more than one optimizers.")
+
+
+class IPEXBF16Precision(PrecisionPlugin):
+    """Create Precision Plugin for IPEX BFloat16."""
+
+    @contextmanager
+    def forward_context(self):
+        """AMP for managing model forward/training_step/evaluation_step/predict_step."""
+        # Using IPEX bf16 and torch.autocast(...) reports a segmentation fault
+        # in PyTorch 1.11.
+        # torch.autocast("cpu", args...) is equivalent to torch.cpu.amp.autocast(args...)
+        # in PyTorch 1.12.
+        with torch.cpu.amp.autocast(dtype=torch.bfloat16):
+            yield
+
+    def optimizer_step(self,
+                       model: Union["pl.LightningModule", Module],
+                       optimizer: Optimizer,
+                       optimizer_idx: int,
+                       closure: Callable[[], Any],
+                       **kwargs: Any) -> Any:
+        """Hook to run the optimizer step."""
+        if type(optimizer) in IPEX_FUSED_OPTIMIZER_LIST:
+            return super().optimizer_step(model, optimizer, optimizer_idx, closure, **kwargs)
+
+        if isinstance(model, pl.LightningModule):
+            closure = partial(self._wrap_closure, model, optimizer, optimizer_idx, closure)
+
+        # Only `torch.optim.LBFGS`  need to reevaluate closure multiple times
+        # in optimizer.step(...) now.
+        if isinstance(optimizer, LBFGS):
+            invalidInputError(False,
+                              "IPEX BFloat16 and the LBFGS optimizer are not compatible "
+                              f"(optimizer {optimizer_idx}")
+
+        # Detect custom optimzer
+        if type(optimizer).__name__ not in dir(torch.optim):
+            warning("Closure use in optimizer.step(...) is not currently supported"
```

Comment:
Users are not likely to understand this closure (he/she did not write any). Maybe we change to an error message more meaningful for the users and also add some suggestions to the user.